src/herbpy/action/rogue.py

```diff
@@ -0,0 +1,354 @@
+import logging, openravepy, prpy 
+from prpy.action import ActionMethod
+from prpy.planning.base import PlanningError
+from contextlib import contextmanager 
+from prpy.util import FindCatkinResource
+import numpy, cPickle, time, os.path
+
+logger = logging.getLogger('herbpy')
+
+@ActionMethod
+def PointAt(robot, focus, manip=None, render=False):
+    """
+    @param robot The robot performing the point
+    @param focus The 3-D coordinate in space or object 
+                 that is being pointed at
+    @param manip The manipulator to perform the point with. 
+                 This must be the right arm
+    @param render Render tsr samples during planning
+    """
+    env = robot.GetEnv()
+    pointing_coord = GetPointFrom(env, focus)
+    Point(robot, pointing_coord, manip, render)
+
+@ActionMethod
+def PresentAt(robot, focus, manip=None, render=True):
+    """
+    @param robot The robot performing the presentation
+    @param focus The 3-D coordinate in space or object that 
+                 is being presented
+    @param manip The manipulator to perform the presentation with. 
+                 This must be the right arm.
+    @param render Render tsr samples during planning
+    """
+    env = robot.GetEnv()
+    presenting_coord = GetPointFrom(env, focus)
+    Present(robot, presenting_coord, manip, render)
+
+@ActionMethod
+def SweepAt(robot, start, end, manip=None, margin=0.3, render=True):
+    """
+    @param robot The robot performing the sweep
+    @param start The object or 3-d position that marks the start
+    @param end The object of 3-d position that marks the end
+    @param manip The manipulator to perform the sweep
+    @param margin The distance between the start object and the hand,
+                  so the vertical space between the hand and objects. 
+                  This must be enough to clear the objects themselves.
+    @param render Render tsr samples during planning
+    """
+    env = robot.GetEnv()
+    start_coord = GetPointFrom(env, start)
+    end_coord = GetPointFrom(env, end)
+    Sweep(robot, start_coord, end_coord, manip, margin, render)
+
+
+def GetPointFrom(env, focus):
+    """
+    @param env The environment where the item exists
+    @param focus The area to be referred to
+    """
+    #Pointing at an object
+    if isinstance(focus, openravepy.KinBody):
+        with env: 
+            focus_trans = focus.GetTransform()
+        coord = list(focus_trans[0:3, 3])
```

Comment:
You should also support `openravepy.KinBody.Link`. I think the same code will work, so you can simply change the condition to:

```yaml
if isinstance(focus, (openravepy.KinBody, openravepy.KinBody.Link)):
```